pkg/apis/eksctl.io/v1alpha5/validation.go

```diff
@@ -585,6 +586,15 @@ func ValidateManagedNodeGroup(ng *ManagedNodeGroup, index int) error {
 		ng.DesiredCapacity = ng.MinSize
 	}
 
+	if ng.UpdateConfig != nil {
+		if ng.UpdateConfig.MaxUnavailable != nil && ng.UpdateConfig.MaxUnavailablePercentage != nil {
+			return fmt.Errorf("cannot use maxUnavailable and maxUnavailablePercentage at the same time")
+		}
+		if aws.IntValue(ng.UpdateConfig.MaxUnavailable) > aws.IntValue(ng.MaxSize) || aws.IntValue(ng.UpdateConfig.MaxUnavailablePercentage) > aws.IntValue(ng.MaxSize) {
```

Comment:
maxUnavailablePercentage can be greater than maxSize of the node group. For ex: a node group with 20 nodes should be able to upgrade 5 nodes at a time which makes it 25% ( greater than 20max size )

Reply:
of course. 🤦 changed it, thanks for spotting this!